### PR TITLE
Adding container support issue #3222

### PR DESCRIPTION
**Pull request by stretch4x4**
Fixes issue #3222
Now templateSelection can have access to the parent container so that
classes can be added for styling

### CODE REVIEW

File: tests/selection/multiple-tests.js [hunk 1]
```diff
@@ -33,6 +33,41 @@ test('display uses templateSelection', function (assert) {
...
+    templateOptions
+  );
+
+  var $container = selection.selectionContainer();
+  
+  var out = selection.display({
+    text: 'test'
+  }, $container);
+
+  for (var i = 0; i < $container[0].classList.length; i += 1) {
```
Comment (kevin-brown) [inline tests/selection/multiple-tests.js]: You should be able to just use `$container[0].classList.contains('testclass')` here. Though that isn't supported in IE8, I'd just use `$container.hasClass('testclass')` honestly.

Comment (stretch4x4) [inline tests/selection/multiple-tests.js]: I did think that was a bit average but was struggling to think of a better way last night.
Thanks for the feedback, also helped me pick up that I had forgotten to rename the variable so it wasn't actually being tested anyway :(

**Comment by kevin-brown** [comment]
The only thing holding me back on merging this is that the single and multiple selection templates now have different parameters. If we can integrate this in with `SingleSelection`, that would be perfect.

**Comment by stretch4x4** [comment]
I didn't really think it was applicable to single select due to it not really using the tag template and it seemed to fall back ok if you didn't use the container but it shouldn't be hard to add to the single selection and no doubt somebody will find a use.
Will see if I can find 5 minutes to solve that.

**Comment by stretch4x4** [comment]
Finally got around to adding the single select support, hopefully that is ready to merge now

**Comment by kevin-brown** [comment]
Looks good, thanks for adding support for the single select as well.

**Comment by stretch4x4** [comment]
No worries

**Pull request merged**